### ISSUE TITLE
[FEATURE REQUEST] Vocabulary mode reading <-> meaning

### ISSUE THREAD

**Issue by omgold** [blastrock/kakugo#49]
I think it would be helpful to be able to learn vocabulary in modes Hiragana to Meaning and Meaning to Hiragana.

At least to me it feels easier and more natural to first learn English<->Hiragana then Hiragana<->Kanji instead of English<->Kanji then Kanji<->Hiragana. I find learning a Kanji by only knowing the meaning as an English word (but not the reading) quite hard.

Alternatively, a mode which shows all 3 of Meaning, Hiragana and Kanji would solve the issue also. To be precise, it would show Hiragana and Kanji together in the same field (one above the other, as child's books do).

**Comment by blastrock** [comment]
I don't think a reading to meaning test is very helpful, a lot of kanji have the same reading, and it's not (IMO) a skill that you will often use when reading or writing Japanese. I want to encourage learners to learn the kanji instead of readings, and associate them with english.

If it can help, the test I practiced the most is the kanji to reading one. And I try to remember the meaning of the kanji before picking an answer. If I can remember the reading, but I have forgotten the meaning, I click on "don't know". I do it this way because I think associating the reading with the kanji is the hardest thing, while associating anything with the english words is much easier. So in the end, this helps me learn both the meaning and reading of kanji.

**Comment by fraengers** [comment]
I totally agree with omgold. Right now it is: English to random symbols that I cannot pronounce, and vice versa. 

My first priority while learning a language is to be able to understand what people say (e.g. when watching TV shows) and then being able to speak it myself (e.g. on a trip). Reading kanji text comes after that, because there is less need for it (e.g. browsing a Japanese website can be done with a translator). The last thing would be (hand)writing with kanji, I don’t know when I would need that.

I'd like to see a vocabulary mode: _meaning to word_, where there is an English word and you have to spell the Japanese word with romaji (which will be then displayed as hiragana) or with a hiragana keyboard of some kind. Or as an option choose a word from a list.

And a mode _word to meaning_, where you read the hiragana word or maybe better, kanji with hiragana on top, and choose from a list of English words.

Adding the hiragana reading above the kanji (as an option) in the current modes would solve most of that. 
But I think there should be an additional mode, where you have to spell it. Maybe like the current _meaning to word_ but with an intermediate step where you have to spell it.
